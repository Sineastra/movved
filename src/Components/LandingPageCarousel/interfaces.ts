export interface slideIntF {
	image: string,
	link: string
}