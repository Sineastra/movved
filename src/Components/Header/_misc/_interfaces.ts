import { signedUserInfoInterface } from "../../../_misc/interfaces"
import React from "react"


export interface desktopPropsInterface {
	loggedUser: signedUserInfoInterface | null,
	className?: string,
}

export interface asideNavPropsInterface {
	loggedUser: signedUserInfoInterface | null,
	className?: string,
}

export interface mobilePropsInterface {
	toggleAsideNav: (event: React.MouseEvent) => void,
	className?: string
}