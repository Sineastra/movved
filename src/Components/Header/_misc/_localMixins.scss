@import "../../../_scssPartials/mixins/mixins";

@mixin aside-nav-link {
    height: 25%;
    text-align: start;
    text-decoration: none;
    @include centeredFlex(row);
    color: var(--theme-text-color);

    &:hover {
        color: var(--theme-secondary-color)
    }
}