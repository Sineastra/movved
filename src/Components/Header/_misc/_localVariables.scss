$aside-nav-height: 300px;
$header-height: 65px;