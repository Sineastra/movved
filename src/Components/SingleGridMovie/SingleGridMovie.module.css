.wrapper {
  width: 100%;
  height: 100%;
}

.slide {
  width: 100%;
  height: 100%;
}

a.link {
  width: 100%;
  height: 100%;
  text-decoration: none;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
  color: var(--theme-text-color);
}
a.link > span {
  position: absolute;
  z-index: 2;
  background-color: rgba(0, 0, 0, 0.9);
  width: 32px;
  height: 25px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  border-radius: 5px;
  color: var(--theme-text-color);
  font-size: 0.7rem;
  font-family: "Monda", sans-serif;
}
a.link > span:nth-of-type(2) {
  right: 5px;
  top: 5px;
}
a.link > span:nth-of-type(3) {
  right: 42px;
  top: 5px;
}
a.link > span:first-of-type {
  left: 5px;
  top: 5px;
}
a.link > h4, a.link h5 {
  margin: 3px 0;
  color: lightgray;
  font-family: "Monda", sans-serif;
}
a.link > h5 {
  color: gray;
}

/*# sourceMappingURL=SingleGridMovie.module.css.map */
