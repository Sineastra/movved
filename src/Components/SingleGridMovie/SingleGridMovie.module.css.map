{"version":3,"sourceRoot":"","sources":["SingleGridMovie.module.scss","../MoviesCategoryCarousel/_localMixins.scss"],"names":[],"mappings":"AAKA;ECJI;EACA;;;ADOJ;ECRI;EACA;EDSA;EACA;EACA;EACA;EACA;EACA;;AAEA;EACI;EACA;EACA;EACA,OAnBK;EAoBL;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;AAEA;EACI;EACA;;AAGJ;EACI;EACA;;AAGJ;EACI;EACA;;AAIR;EACI;EACA;EACA;EACA;EACA;;AAGJ;EACI","file":"SingleGridMovie.module.css"}