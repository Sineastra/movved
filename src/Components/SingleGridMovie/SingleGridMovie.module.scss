@import "../../_scssPartials/mixins/mixins";

$span-width: 32px;
@mixin slide-width-height {
    width: 200px;
    height: 300px;
}

.slide {
    @include slide-width-height;
}

a.link {
    @include slide-width-height;
    text-decoration: none;
    position: absolute;
    top: 0;
    left: 0;
    z-index: 1;
    color: var(--theme-text-color);

    & > span {
        position: absolute;
        z-index: 2;
        background-color: rgba(0, 0, 0, 0.9);
        width: $span-width;
        height: 25px;
        display: inline-flex;
        justify-content: center;
        align-items: center;
        border-radius: 5px;
        color: var(--theme-text-color);
        font-size: 0.7rem;
        font-family: 'Monda', sans-serif;

        &:nth-of-type(2) {
            right: 5px;
            top: 5px;
        }

        &:nth-of-type(3) {
            right: $span-width + 10;
            top: 5px;
        }

        &:first-of-type {
            left: 5px;
            top: 5px;
        }
    }

    & > h4, h5 {
        margin: 0;
        margin-top: 3px;
        margin-bottom: 3px;
        color: lightgray;
        font-family: 'Monda', sans-serif;
    }

    & > h5 {
        color: gray;
    }
}
