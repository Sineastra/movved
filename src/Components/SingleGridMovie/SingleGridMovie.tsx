import { movieInterface } from "../../_misc/interfaces"
import { Link } from "react-router-dom"
import styles from "./SingleGridMovie.module.scss"


interface propsIntF {
	movie: movieInterface,
	className?: string,
}

const SingleGridMovie = ({ movie, className }: propsIntF) => {
	return (
		<div className={ className }>
			<Link to={ `/movie/${ movie.watchLink }` } className={ styles.link }>
				<span className={ styles.yearSpan }>{ movie.year }</span>
				{ movie.isDubbed && <span className={ styles.subSpan }>Dub</span> }
				{ movie.isSubbed && <span className={ styles.dubSpan }>Sub</span> }
				<img src={ movie.smallPoster } alt="poster" className={ styles.slide }/>
				<h4 role="title">{ movie.title }</h4>
				<h5 role="eng-title">{ movie.engTitle }</h5>
				<h5 role="genre">{ movie.genres }</h5>
			</Link>
		</div>
	)
}

export default SingleGridMovie