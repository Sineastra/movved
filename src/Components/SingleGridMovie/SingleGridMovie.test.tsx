import { movie } from "../../_misc/interfaces"
import { BrowserRouter } from "react-router-dom"
import SingleGridMovie from "./SingleGridMovie"
import renderer from "react-test-renderer"


const mockedMovie: movie = {
	_id: '0',
	title: 'a',
	smallPoster: 'b',
	bigPoster: 'c',
	isSubbed: true,
	isDubbed: true,
	year: 0,
	watchLink: 'd',
	engTitle: 'e',
	genre: 'f',
}

const renderScreen = (movie: movie) =>
	renderer.create(
		<BrowserRouter>
			<SingleGridMovie movie={ movie }/>
		</BrowserRouter>,
	).toJSON()

describe('---> Testing Single Slide of Movies Category Carousel', () => {
	it("renders correctly when isSubbed and/or isDubbed are present (snapshot testing)", () => {
		const tree = renderScreen(mockedMovie)

		expect(tree).toMatchSnapshot()
	})
	it("renders correctly when isSubbed and/or isDubbed are NOT present (snapshot testing)", () => {
		const modifMockedMovie = { ...mockedMovie, ...{ isSubbed: false, isDubbed: false } }
		const tree = renderScreen(modifMockedMovie)

		expect(tree).toMatchSnapshot()
	})
})