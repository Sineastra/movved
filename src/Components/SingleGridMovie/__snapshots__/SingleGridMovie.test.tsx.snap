// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`---> Testing Single Slide of Movies Category Carousel renders correctly when isSubbed and/or isDubbed are NOT present (snapshot testing) 1`] = `
<div>
  <a
    className="link"
    href="/movie/d"
    onClick={[Function]}
  >
    <span
      className="yearSpan"
    >
      0
    </span>
    <img
      alt="poster"
      className="slide"
      src="b"
    />
    <h4
      role="title"
    >
      a
    </h4>
    <h5
      role="eng-title"
    >
      e
    </h5>
    <h5
      role="genre"
    >
      f
    </h5>
  </a>
</div>
`;

exports[`---> Testing Single Slide of Movies Category Carousel renders correctly when isSubbed and/or isDubbed are present (snapshot testing) 1`] = `
<div>
  <a
    className="link"
    href="/movie/d"
    onClick={[Function]}
  >
    <span
      className="yearSpan"
    >
      0
    </span>
    <span
      className="subSpan"
    >
      Dub
    </span>
    <span
      className="dubSpan"
    >
      Sub
    </span>
    <img
      alt="poster"
      className="slide"
      src="b"
    />
    <h4
      role="title"
    >
      a
    </h4>
    <h5
      role="eng-title"
    >
      e
    </h5>
    <h5
      role="genre"
    >
      f
    </h5>
  </a>
</div>
`;
